From 7ab98c333b1d9e414f15ea0afd52f3fa4f4ff9d9 Mon Sep 17 00:00:00 2001
Message-Id: <7ab98c333b1d9e414f15ea0afd52f3fa4f4ff9d9.1582951866.git.devel@agrez.net>
From: Vaughan Agrez <devel@agrez.net>
Date: Sat, 29 Feb 2020 12:48:10 +0800
Subject: [PATCH] perf build fix for epel8

Revert upstream patch: perf arm64: Fix include path for
asm-generic/unistd (commit: 5ab1de932e2923f490645ad017a689c5b58dc433)
---
 tools/perf/arch/arm64/Makefile                    | 5 ++---
 tools/perf/arch/arm64/entry/syscalls/mksyscalltbl | 6 +++---
 2 files changed, 5 insertions(+), 6 deletions(-)

diff --git a/tools/perf/arch/arm64/Makefile b/tools/perf/arch/arm64/Makefile
index dbef716a1..f013b115d 100644
--- a/tools/perf/arch/arm64/Makefile
+++ b/tools/perf/arch/arm64/Makefile
@@ -11,8 +11,7 @@ PERF_HAVE_ARCH_REGS_QUERY_REGISTER_OFFSET := 1
 
 out    := $(OUTPUT)arch/arm64/include/generated/asm
 header := $(out)/syscalls.c
-incpath := $(srctree)/tools
-sysdef := $(srctree)/tools/arch/arm64/include/uapi/asm/unistd.h
+sysdef := $(srctree)/tools/include/uapi/asm-generic/unistd.h
 sysprf := $(srctree)/tools/perf/arch/arm64/entry/syscalls/
 systbl := $(sysprf)/mksyscalltbl
 
@@ -20,7 +19,7 @@ systbl := $(sysprf)/mksyscalltbl
 _dummy := $(shell [ -d '$(out)' ] || mkdir -p '$(out)')
 
 $(header): $(sysdef) $(systbl)
-	$(Q)$(SHELL) '$(systbl)' '$(CC)' '$(HOSTCC)' $(incpath) $(sysdef) > $@
+	$(Q)$(SHELL) '$(systbl)' '$(CC)' '$(HOSTCC)' $(sysdef) > $@
 
 clean::
 	$(call QUIET_CLEAN, arm64) $(RM) $(header)
diff --git a/tools/perf/arch/arm64/entry/syscalls/mksyscalltbl b/tools/perf/arch/arm64/entry/syscalls/mksyscalltbl
index 2dbb8cade..52e197317 100755
--- a/tools/perf/arch/arm64/entry/syscalls/mksyscalltbl
+++ b/tools/perf/arch/arm64/entry/syscalls/mksyscalltbl
@@ -11,8 +11,7 @@
 
 gcc=$1
 hostcc=$2
-incpath=$3
-input=$4
+input=$3
 
 if ! test -r $input; then
 	echo "Could not read input file" >&2
@@ -29,6 +28,7 @@ create_table_from_c()
 
 	cat <<-_EoHEADER
 		#include <stdio.h>
+		#define __ARCH_WANT_RENAMEAT
 		#include "$input"
 		int main(int argc, char *argv[])
 		{
@@ -42,7 +42,7 @@ create_table_from_c()
 	printf "%s\n" "	printf(\"#define SYSCALLTBL_ARM64_MAX_ID %d\\n\", __NR_$last_sc);"
 	printf "}\n"
 
-	} | $hostcc -I $incpath/include/uapi -o $create_table_exe -x c -
+	} | $hostcc -o $create_table_exe -x c -
 
 	$create_table_exe
 
-- 
2.21.1

